activerecord/lib/arel/insert_manager.rb

```diff
@@ -38,6 +38,11 @@ def insert(fields)
       self
     end
 
+    def comment(*values)
+      @ast.comment = Nodes::Annotation.new(*values)
+      self
+    end
+
```

Comment:
Seems the AST node is an `annotation`, since it's a specific usage of a comment that always appears at the end of the query.

Reply:
I see what you're saying. Perhaps if `Annotation` is the mixin and we're instantiating a `Comment` here then this property name `comment` makes sense.

Reply:
(At least, I _think_ I see what you're saying. 🙂 If it seems like I've misinterpreted you please let me know.)

Something that may be clear by inference but I'll make explicit, the naming at the `AR::Relation` level around "annotate" / "annotation" was chosen primarily because of the likely number of `:comment` / `:comments` associations and scopes in the wild, and wanting to avoid ambiguity or collision there.

At the Arel AST level, in terms of the SQL domain, the term "comment" seems the most precise. Hence the intent to switch back to a `Comment` concrete node class and name the more generic mixin `Annotation`.

Reply:
As of the latest code, there are now `Arel::Nodes::Comment` and `Arel::Nodes::OptimizerHints` classes that each mix in `Arel::Nodes::Annotation`. I think that naming makes the most sense in the SQL domain, but if you disagree I can change it around.